cattle/utils.py

```diff
@@ -73,6 +77,42 @@ def unwrap(json_object):
         return json_object
 
 
+class CadvisorAPIClient(object):
+    def __init__(self, host, port, version='v1.2', proto='http://'):
+        self.url = proto + host + ':' + str(port) + '/api/{0}'.format(version)
+
+    def get_containers(self):
+        return self._get(self.url + '/containers')
+
+    def get_latest_stat(self):
+        return self.get_containers()['stats'][-1]
+
+    def get_stats(self):
+        return self.get_containers()['stats']
+
+    def timestamp_diff(self, time_current, time_prev):
+        time_current_conv = self._timestamp_convert(time_current)
+        time_prev_conv = self._timestamp_convert(time_prev)
+
+        diff = (time_current_conv - time_prev_conv).total_seconds()
+        return round((diff * 10**9))
+
+    def _timestamp_convert(self, stime):
+        # This is a bit hacky
+        t_strip = stime[:-4]
+        return datetime.strptime(t_strip, "%Y-%m-%dT%H:%M:%S.%f")
+
+    def _marshall_to_python(self, data):
+        if isinstance(data, str):
+            return json.loads(data)
+
+    def _get(self, url):
+        resp = urllib2.urlopen(url)
```

Comment:
the urllib2.urlopen doesn't return an object with an __exit__ method so the 'with' context manager doesn't work in this case.